shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/SchemaBuilder.java

```diff
@@ -39,42 +36,30 @@
 public final class SchemaBuilder {
     
     /**
-     * Build sharding sphere schema.
+     * Build schema for kernel.
      *
-     * @param materials schema builder materials
+     * @param tableMetaDatas table meta datas
+     * @param rules sharding sphere rule
      * @return sharding sphere schema
-     * @throws SQLException SQL exception
      */
-    public static ShardingSphereSchema build(final SchemaBuilderMaterials materials) throws SQLException {
-        Collection<String> allTableNames = materials.getRules().stream().filter(each -> each instanceof TableContainedRule)
-                .flatMap(shardingSphereRule -> ((TableContainedRule) shardingSphereRule).getTables().stream()).collect(Collectors.toSet());
-        return new ShardingSphereSchema(TableMetaDataBuilder.load(allTableNames, materials));
+    public static ShardingSphereSchema buildForKernel(final Collection<TableMetaData> tableMetaDatas, final Collection<ShardingSphereRule> rules) {
```

Comment:
@tuichenchuxin Do you think `buildKernelSchema` and `buildFederateSchema` is better?